explorer/testdata/var/local/reference_expression.carbon

```diff
@@ -0,0 +1,84 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// CHECK:STDOUT: Initialize c1 from reference expression
+// CHECK:STDOUT: 0: Heap{}, 1: C{.id = 1}, 2: C{.id = 1}
+// CHECK:STDOUT: c1 destroyed
+// CHECK:STDOUT: c1 destroyed
+// CHECK:STDOUT:
+// CHECK:STDOUT: Initialize c2 from value expression
+// CHECK:STDOUT: 0: Heap{}, 1: !!C{.id = 1}, 2: !!C{.id = 1}, 3: C{.id = 2}, 4: C{.id = 2}
```

Comment:
You might want to print the heap between each function, so that the scope destructions are easily ignored. You could do that either with a call in Main, or just write a class with a destructor that prints the allocations and allocate it before other local vars.